ledger/narwhal/batch-header/src/bytes.rs

```diff
@@ -34,16 +34,24 @@ impl<N: Network> FromBytes for BatchHeader<N> {
         let timestamp = i64::read_le(&mut reader)?;
 
         // Read the number of transmission IDs.
-        let num_transmissions = u32::read_le(&mut reader)?;
+        let num_transmission_ids = u32::read_le(&mut reader)?;
+        // Ensure the number of transmission ids is within bounds.
+        if num_transmission_ids > i32::MAX as u32 {
+            return Err(error("Number of transmission ids exceeds maximum"));
+        }
         // Read the transmission IDs.
-        let mut transmission_ids = IndexSet::new();
-        for _ in 0..num_transmissions {
+        let mut transmission_ids = IndexSet::with_capacity(num_transmission_ids as usize);
```

Comment:
```suggestion
        let mut transmission_ids = IndexSet::new();
```
it's safer, especially if someone sends multiple messages that suggest the maximum length (even if the contents don't hold that many)